crates/matrix-sdk-sqlite/src/lib.rs

```diff
@@ -43,13 +44,21 @@ pub use self::state_store::{SqliteStateStore, DATABASE_NAME as STATE_STORE_DATAB
 #[cfg(test)]
 matrix_sdk_test_utils::init_tracing_for_tests!();
 
+#[derive(Clone, Debug, Eq, PartialEq, Zeroize)]
+pub enum Secret {
+    #[zeroize]
+    Key([u8; 32]),
```

Comment:
This should be inside a `Box` or inside a `Zeroizing` struct so we avoid unintended copies due to moves being a `memcpy` call in Rust land. More info here: https://benma.github.io/2020/10/16/rust-zeroize-move.html.

In short, every time you pass the `Secret` to a function a `memcpy` call will copy the contents of the array.

If a `Box` is used, the pointer is instead copied.